packages/amplify-category-storage/provider-utils/awscloudformation/index.js

```diff
@@ -31,5 +31,18 @@ function updateResource(context, category, service) {
   return updateWalkthrough(context, defaultValuesFilename, serviceMetadata);
 }
 
+function migrateResource(projectPath, service, resourceName) {
+  const serviceMetadata = JSON.parse(fs.readFileSync(`${__dirname}/../supported-services.json`))[service];
+  const { serviceWalkthroughFilename } = serviceMetadata;
+  const serviceWalkthroughSrc = `${__dirname}/service-walkthroughs/${serviceWalkthroughFilename}`;
+  const { migrate } = require(serviceWalkthroughSrc);
+
+  if (!migrate) {
+    console.log(`No migration required for ${resourceName}`);
+  }
+
+  return migrate(projectPath, resourceName);
```

Comment:
if migrate can be undefined or null, then this line needs to be combined with the previous if block, call migrate only if it's defined.